app/controllers/katello/concerns/smart_proxies_controller_extensions.rb

```diff
@@ -1,68 +0,0 @@
-module Katello
-  module Concerns
-    module SmartProxiesControllerExtensions
-      extend ActiveSupport::Concern
-
-      module Overrides
-        def show
-          @task_search_url = main_app.foreman_tasks_tasks_path(:search => "resource_id = #{@smart_proxy.id} AND resource_type = #{@smart_proxy.class}")
-          render 'foreman/smart_proxies/show', :layout => 'katello/layouts/foreman_with_bastion'
-        end
-
-        def action_permission
-          if params[:action] == 'pulp_status' || params[:action] == 'pulp_storage'
-            :view
-          else
-            super
-          end
-        end
-      end
-
-      included do
-        prepend Overrides
-        helper 'bastion/layout'
-
-        append_view_path('app/views/foreman')
-        before_action :find_resource_and_status, :only => [:pulp_storage, :pulp_status]
-
-        def pulp_storage
-          @storage = @smart_proxy.pulp_disk_usage
```

Comment:
I'm not sure how much of this file is still used, but this bit definitely is still used to show the Pulp 3 storage amount. `@storage` being `nil` in the error I pasted earlier makes sense since this file was deleted.